utils/src/big_bit_ops.rs

```diff
@@ -0,0 +1,108 @@
+//! This module provides a set of functions to perform bit operations on big integers.
+//! In particular, it gives XOR and NOT for BigUint.
+use num_bigint::BigUint;
+use rand::Rng;
+use std::cmp::Ordering;
+
+/// Exclusive or of the bits of two BigUint inputs
+pub fn big_xor(input1: &BigUint, input2: &BigUint) -> BigUint {
```

Comment:
Or maybe it's possible to implement this one.
https://doc.rust-lang.org/std/ops/trait.BitXor.html
or perhaps it's already implemented for `BigUint`.

Reply:
Mmm I like the idea, I will try

Reply:
Okay, I will have a trait for all of these, but the one inside the std library requires to clone the self because you cannot copy biguint, thus made a new one